0.8.2: fails to build with -Dwith_wayland=disabled

**Do not report issue for old MangoHud versions**

**Describe the bug**
Trying to build with the wayland feature disabled `-Dwith_wayland=disabled` results in the build failure:

```
[...]
[54/95] x86_64-pc-linux-gnu-c++ -Isrc/libMangoHud_opengl.so.p -Isrc -I../MangoHud-v0.8.2/src -I../MangoHud-v0.8.2/include -Isubprojects/imgui-1.89.9 -I../MangoHud-v0.8.2/subprojects/imgui-1.89.9 -I../MangoHud-v0.8.2/subprojects/imgui-1.89.9/backends -I../MangoHud-v0.8.2/subprojects/Vulkan-Headers-1.2.158/include -Isubprojects/implot-0.16 -I../MangoHud-v0.8.2/subprojects/implot-0.16 -I. -I/usr/x86_64-pc-linux-gnu/include/dbus-1.0 -I/usr/x86_64-pc-linux-gnu/lib/dbus-1.0/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -O3 -Werror=missing-declarations -Werror=return-type -Wno-unused-parameter -fno-math-errno -fno-trapping-math -Wno-non-virtual-dtor -Wno-missing-field-initializers -march=native -O2 -pipe -march=native -O2 -fPIC -pthread -DSPDLOG_SHARED_LIB -DSPDLOG_FMT_EXTERNAL '-DIMGUI_API=__attribute__((visibility("default")))' -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS '-DPACKAGE_VERSION="v0.8.2"' -DSPDLOG_COMPILED_LIB -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_DEBUG -D_GNU_SOURCE -DHAVE_PTHREAD -DHAVE_FTRACE -DUSE_GCC_ATOMIC_BUILTINS -DHAVE_TIMESPEC_GET -DHAVE___BUILTIN_BSWAP32 -DHAVE___BUILTIN_BSWAP64 -DHAVE___BUILTIN_CLZ -DHAVE___BUILTIN_CLZLL -DHAVE___BUILTIN_CTZ -DHAVE___BUILTIN_EXPECT -DHAVE___BUILTIN_FFS -DHAVE___BUILTIN_FFSLL -DHAVE___BUILTIN_POPCOUNT -DHAVE___BUILTIN_POPCOUNTLL -DHAVE___BUILTIN_UNREACHABLE '-DMANGOHUD_ARCH="64bit"' -DHAVE_X11 -DHAVE_DBUS -DVK_USE_PLATFORM_XLIB_KHR -MD -MQ src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o -MF src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o.d -o src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o -c ../MangoHud-v0.8.2/src/gl/inject_egl.cpp
FAILED: [code=1] src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o 
x86_64-pc-linux-gnu-c++ -Isrc/libMangoHud_opengl.so.p -Isrc -I../MangoHud-v0.8.2/src -I../MangoHud-v0.8.2/include -Isubprojects/imgui-1.89.9 -I../MangoHud-v0.8.2/subprojects/imgui-1.89.9 -I../MangoHud-v0.8.2/subprojects/imgui-1.89.9/backends -I../MangoHud-v0.8.2/subprojects/Vulkan-Headers-1.2.158/include -Isubprojects/implot-0.16 -I../MangoHud-v0.8.2/subprojects/implot-0.16 -I. -I/usr/x86_64-pc-linux-gnu/include/dbus-1.0 -I/usr/x86_64-pc-linux-gnu/lib/dbus-1.0/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -O3 -Werror=missing-declarations -Werror=return-type -Wno-unused-parameter -fno-math-errno -fno-trapping-math -Wno-non-virtual-dtor -Wno-missing-field-initializers -march=native -O2 -pipe -march=native -O2 -fPIC -pthread -DSPDLOG_SHARED_LIB -DSPDLOG_FMT_EXTERNAL '-DIMGUI_API=__attribute__((visibility("default")))' -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS '-DPACKAGE_VERSION="v0.8.2"' -DSPDLOG_COMPILED_LIB -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_DEBUG -D_GNU_SOURCE -DHAVE_PTHREAD -DHAVE_FTRACE -DUSE_GCC_ATOMIC_BUILTINS -DHAVE_TIMESPEC_GET -DHAVE___BUILTIN_BSWAP32 -DHAVE___BUILTIN_BSWAP64 -DHAVE___BUILTIN_CLZ -DHAVE___BUILTIN_CLZLL -DHAVE___BUILTIN_CTZ -DHAVE___BUILTIN_EXPECT -DHAVE___BUILTIN_FFS -DHAVE___BUILTIN_FFSLL -DHAVE___BUILTIN_POPCOUNT -DHAVE___BUILTIN_POPCOUNTLL -DHAVE___BUILTIN_UNREACHABLE '-DMANGOHUD_ARCH="64bit"' -DHAVE_X11 -DHAVE_DBUS -DVK_USE_PLATFORM_XLIB_KHR -MD -MQ src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o -MF src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o.d -o src/libMangoHud_opengl.so.p/gl_inject_egl.cpp.o -c ../MangoHud-v0.8.2/src/gl/inject_egl.cpp
../MangoHud-v0.8.2/src/gl/inject_egl.cpp: In function 'int eglTerminate(void*)':
../MangoHud-v0.8.2/src/gl/inject_egl.cpp:244:9: error: 'wayland_data_unref' was not declared in this scope
  244 |         wayland_data_unref(NULL, display);
      |         ^~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

Complete build log: [MangoHud-0.8.2_build.log](https://github.com/user-attachments/files/23827750/MangoHud-0.8.2_build.log)


**List relevant hardware/software information**
-

**To Reproduce**
Steps to reproduce the behavior:

1. Try to build with -Dwith_wayland=disabled
2. See error

**Expected behavior**
Build without running into build failures.

**Screenshots**
-

**Additional context**
-

--- a/src/gl/inject_egl.cpp
+++ b/src/gl/inject_egl.cpp
@@ -239,10 +239,12 @@ EXPORT_C_(int) eglTerminate(void *display)
 
     ret = pfn_eglTerminate(display);
 
+#ifdef HAVE_WAYLAND
     if (ret)
     {
         wayland_data_unref(NULL, display);
     }
+#endif
 
     return ret;
 }

